classes/local/factor/object_factor.php

```diff
@@ -36,6 +36,7 @@ public function define_add_factor_form_definition_after_data($mform);
     public function define_login_form($mform);
     public function validation($data);
     public function add_user_factor($data);
-    public function get_user_factors($user);
+    public function get_all_user_factors($user);
+    public function get_enabled_user_factor($user);
```

Comment:
I think this should be passed a $user object, not a $userid

Reply:
Removed argument at all. Using USER global variable in methods